tools/regen_gh_pages.sh

```diff
@@ -0,0 +1,132 @@
+
+        # Extract the javadocs
+        pushd "$work_dir/RPMS"
+            _e rpm2cpio jss-javadoc*.rpm > javadoc.cpio
+            _e cpio -idv < javadoc.cpio
+            _e cp -rv ./usr/share/javadoc/jss* "$output_dir"
+        popd # "$work_dir"
+
+        rm -rf "$work_dir"
+    fi
+}
+
+function build_doc() {
+    local output_dir="$1"
+
+    if [ -e docs/ ]; then
+        _e cp -rv docs "$output_dir"
+    fi
+}
+
+function clean_checkout() {
+    local branch="$1"
+
+    _e git clean -xdf
+    _e git checkout "$branch"
+}
+
+function main() {(
+    set -e
+
+    local user="$1"
+    local remote="https://github.com/$user/jss"
+    local upstream="https://github.com/dogtagpki/jss"
+
+    local sandbox="/tmp/jss-docs-sandbox"
+    local artifacts="/tmp/jss-docs-artifacts"
+
+    if [ -z "$user" ]; then
+        echo "Usage: update-gh-pages.sh <username>"
+        echo "Update the user's gh-pages branch with the latest javadocs content."
+        return 1
+    fi
+
+    # Start with a fresh environment each time.
+    _e rm -rf "$sandbox" "$artifacts"
+    _e mkdir -p "$sandbox"
+    _e mkdir -p "$artifacts"
+
+    pushd "$sandbox"
+        _e git clone "$remote" jss
+        cd jss
+        _e git remote add upstream "$upstream"
+        _e git fetch --all
+
+        # Build docs for master branch
+        clean_checkout "upstream/master"
+        build_javadoc "$artifacts/javadocs-master"
+        build_doc "$artifacts/docs-master"
+
+        # Build docs for the v4.5.x branch
+        clean_checkout "upstream/v4.5.x"
+        build_javadoc "$artifacts/javadocs-v4.5.x"
+        build_doc "$artifacts/docs-v4.5.x"
+
+        # Build docs for the v4.4.x branch
+        clean_checkout "upstream/v4.4.x"
+        build_javadoc "$artifacts/javadocs-v4.4.x"
+        build_doc "$artifacts/docs-v4.4.x"
+
+        # Get to the gh-pages branch, in a clean state
+        clean_checkout "upstream/gh-pages"
+        _e git branch -D "gh-pages" || true
+        _e git checkout -b "gh-pages"
+        _e rm -rf javadoc master/javadocs master/docs v4.5.x/javadocs v4.5.x/docs v4.4.x/javadocs v4.4.x/docs
+
+        # Preserve new changes
+        _e mkdir -p master v4.5.x v4.4.x || true
+        _e cp -rv "$artifacts/javadocs-master" master/javadocs || true
+        _e cp -rv "$artifacts/docs-master" master/docs || true
+        _e cp -rv "$artifacts/javadocs-v4.5.x" v4.5.x/javadocs || true
+        _e cp -rv "$artifacts/docs-v4.5.x" v4.5.x/docs || true
+        _e cp -rv "$artifacts/javadocs-v4.4.x" v4.4.x/javadocs || true
+        _e cp -rv "$artifacts/docs-v4.4.x" v4.4.x/docs || true
+
+        # Track all of our new changes
+        _e git add --all
+        _e git commit -m "Update javadocs from master at $(date '+%Y-%m-%d %H:%M')"
+        _e git push --set-upstream origin gh-pages --force
+    popd # "$sandbox"
+
+    _e rm -rf "$sandbox"
+
+    echo ""
+    echo ""
+    echo "All done! To open a PR, click the following link:"
```

Comment:
Once the script is executed successfully on the CI, i see that it'll force push it to **it's** forked version. :+1: 

Does it create a PR automatically or should we click on the link? If you think you can automate this step too, you can take a look at [github hub](https://hub.github.com/)

Reply:
Currently the PR is manual, especially since I like to review it and as you can see in #165, add additional commits. The side effect is that anything which triggers a build triggers an automatic nuke of the branch; perhaps this is fine, but we could fix this by building into a branch-specific GH pages branch.

And yes, if we place this (or a version of it) in the automation pipeline, I'd definitely go ahead and open the PR automatically. I think we can do it with `curl` via the [GitHub API](https://developer.github.com/v3/pulls/#create-a-pull-request), instead of requiring that `hub` be installed on all CI systems.